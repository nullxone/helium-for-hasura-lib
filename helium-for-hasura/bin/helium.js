#!/usr/bin/env node

const path = require("path");
const fs = require("fs-extra");

const { Command } = require("commander");
const program = new Command();

const package = require("../package.json");

program
  .name("helium")
  .description(package.description)
  .version(package.version);

program
  .command("compile")
  .description("Compile Helium models to Hasura metadata")
  .option("-s, --source <string>", "Source Helium Directory", "./helium")
  .option("-t, --target <string>", "Target Metadata Directory", "./metadata")
  .action((options) => {
    require("../metadata").compile(
      path.join(process.cwd(), options.source),
      path.join(process.cwd(), options.target)
    );
  });

program.command("deploy", "Deploy compiled metadata to Hasura", {
  executableFile: "helium-deploy.sh",
});

program
  .command("module")
  .description("Import a module, currently only hasura-auth")
  .command("import")
  .command("hasura-auth")
  .option("-s, --source <string>", "Source Helium Directory", "./helium")
  .description("Import Hasura Auth module to your project")
  .action((options) => {
    fs.copySync(
      path.join(__dirname, "../modules/hasura-auth"),
      path.join(process.cwd(), options.source, "modules/hasura-auth")
    );
  });

program.parse();
