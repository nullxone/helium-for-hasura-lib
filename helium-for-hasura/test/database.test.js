const Database = require("../database");
const { Model } = require("../model");
const path = require("path");
const fs = require("fs-extra");

test("database config to database metadata", () => {
  const appDir = path.join(__dirname, "./AppSimple");
  fs.emptyDirSync(path.join(appDir, "metadata"));

  Database.compile(
    require(path.join(appDir, "helium/Database")),
    path.join(appDir, "metadata")
  );

  const outFile = path.join(appDir, "metadata/databases/databases.yaml");

  expect(fs.existsSync(outFile)).toBe(true);
  expect(fs.readFileSync(outFile, "utf8")).toMatchSnapshot();
});

test("write all tables file", () => {
  const appDir = path.join(__dirname, "./AppSimple");
  const metadataDir = path.join(appDir, "metadata");
  fs.emptyDirSync(metadataDir);

  Model.compile(require(path.join(appDir, "helium/models/Users")), metadataDir);

  Model.compile(require(path.join(appDir, "helium/models/Tasks")), metadataDir);

  Database.writeAllTables(metadataDir);

  const outFile = path.join(
    metadataDir,
    "databases/default/tables/tables.yaml"
  );

  expect(fs.existsSync(outFile)).toBe(true);
  expect(fs.readFileSync(outFile, "utf8")).toMatchSnapshot();
});
