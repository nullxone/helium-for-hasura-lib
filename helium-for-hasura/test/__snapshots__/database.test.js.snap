// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`database config to database metadata 1`] = `
"- name: default
  kind: postgres
  configuration:
    connection_info:
      database_url: test_database_url
      isolation_level: read-committed
      use_prepared_statements: false
  tables: "!include default/tables/tables.yaml"
"
`;

exports[`write all tables file 1`] = `
"- '!include public_tasks.yaml'
- '!include public_users.yaml'
"
`;
